docs/DafnyRef/Topics.md

```diff
@@ -34,6 +34,19 @@ const m := set x | n in {-1, 0, 1, 2} && f(n) >= 1
 const m := set x | n in {-1, 0, 1, 2} && g(n) >= 1
 ```
 
+### 23.2.2. Ghost subset types in comprehensions
+
+A ghost [subset type](#sec-subset-types) is a subset type where one of its constraint is ghost.
+For comprehensions ([forall and exists expressions](#sec-quantifier-expression), [set comprehensions](#sec-set-comprehension-expression) and [map comprehensions](#sec-map-comprehension-expression)), variables with a ghost subset type are not problem if they are in a ghost context.
+However, in a compiled context, things are more tricky.
+
+Consider a comprehension like `set c: GhostSubsetType | c in Collection && P(c) :: c` in a compiled context where `Collection: set<T>` and the constraint of `GhostSubsetType` may not be checked at run-time, because it's ghost. Since Dafny cannot emit code to test the constraint at run-time, Dafny needs to verify it statically.
```

Comment:
I read

> where `Collection: set<T>` and the constraint of `GhostSubsetType` may not be checked at run-time, because it's ghost

as if both `Collection: set<T>` and the constraint of `GhostSubsetType` may not be checked at run-time. Consider changing it to 
> where Collection has type `set<T>` and the constraint of `GhostSubsetType` is ghost so it may not be checked at run-time.